### ISSUE TITLE
Error when using nested variables

### ISSUE THREAD

**Issue by febLey** [sasstools/sass-lint#593]
I'm getting fatal error messages when using variables nested into a selector:

``` sass
.form
  $label-width: 33%
```

In this case I get: 

> Please check validity of the block  Fatal

``` sass
.teaserbox
  $self: &
```

And in this case:

> Please check validity of the block starting from line #2  Fatal

Here's my config

``` yaml
## Documentation
# https://github.com/sasstools/sass-lint/tree/master/docs/rules

## Linter Options
options:
  merge-default-rules: false

## Rule Configuration
rules:
  bem-depth:
    - 1
    -
      max-depth: 3
  border-zero:
   - 2
   -
      convention: 'none'
  extends-before-mixins: 2
  extends-before-declarations: 2
  hex-length: 2
  hex-notation:
    - 2
    -
      style: lowercase
  # indentation:
  #   - 2
  #   -
  #     size: 2
  leading-zero:
    - 2
    -
      include: true
  mixins-before-declarations:
    - 2
    -
      exclude:
        - media
  no-color-keywords: 2
  no-color-literals:
    - 2
    -
      allow-rgba: true
  no-css-comments: 2
  no-debug: 1
  no-duplicate-properties:
    - 2
    # -
    #   exclude:
    #     - display
  no-empty-rulesets: 1
  no-ids: 1
  no-important: 2
  no-invalid-hex: 2
  no-mergeable-selectors:
    - 2
    # -
    #   whitelist:
    #     -
  no-misspelled-properties: 2

  no-qualifying-elements:
    - 2
    -
      allow-element-with-attribute: true
  no-trailing-zero: 2
  no-url-protocols: 2
  no-vendor-prefixes:
    - 1
    # -
    #   excluded-identifiers:
    #     -
  no-warn: 1
  one-declaration-per-line: 2
  placeholder-in-extend: 2
  property-sort-order:
    - 1
    -
      order: 'recess'
      ignore-custom-properties: true
  property-units:
    - 2
    -
      global:
        - rem
        - vh
        - vw
  quotes:
    - 2
    -
      style: double
  shorthand-values: 2
  space-after-bang: 2
  space-after-colon: 2
  space-after-comma: 2
  space-around-operator: 2
  space-before-bang: 2
  space-before-brace: 2
  space-before-colon: 2
  space-between-parens: 2
  trailing-semicolon: 2
  url-quotes: 2
  zero-unit: 2
  class-name-format:
    - 2
    -
      allow-leading-underscore: false
      convention: 'hyphenatedbem'
      ignore:
        - mod_cssanimations
        - mod_csstransforms
        - mod_supports
        - mod_csstransforms3d
        - mod_csstransitions
  function-name-format:
    - 2
    -
      allow-leading-underscore: false
  id-name-format:
    - 2
    -
      allow-leading-underscore: false
  mixin-name-format:
    - 2
    -
      allow-leading-underscore: false
  placeholder-name-format:
    - 2
    -
      allow-leading-underscore: false
  variable-name-format:
    - 2
    -
      allow-leading-underscore: false

```

**Comment by DanPurdy** [comment]
This is to do with the AST that's being used. [Gonzales-PE](https://github.com/tonyganch/gonzales-pe) Your first example will work in our 1.6 update when it's released as it includes the latest version of said AST, your second example Isn't I'm afraid so you'll need to have a look over on the AST for the related issue about assigning the ampersand to a variable.

Unfortunately it's out of our control.

**Comment by febLey** [comment]
It's fixed in version 3.3.4. Will we see an update soon?

**Comment by DanPurdy** [comment]
Yep, we were just waiting to see if some of our other PR's into gonzales would be merged, they need a little bit of work though so we'll get them in at some other point. 

We're just gonna go ahead with our 1.8 release anyway once our outstanding tasks are merged as we have a few great additions to release with it. Should hopefully see it in the next few days.
